### DISCUSSION TITLE
ModuleNotFoundError: No module named 'pywintypes'

### DISCUSSION THREAD

**Discussion by ycw01** [beeware/briefcase#1705] [Q&A]
### Describe the bug

The 'briefcase run' operation encountered an error：
  File "G:\beeware-tutorial\bydkmipwd\build\bydkmipwd\windows\app\src\app\bydkmipwd\MacUtils.py", line 2, in <module>
    import wmi
  File "G:\beeware-tutorial\bydkmipwd\build\bydkmipwd\windows\app\src\app_packages\wmi.py", line 105, in <module>
    from win32com.client import GetObject, Dispatch
  File "G:\beeware-tutorial\bydkmipwd\build\bydkmipwd\windows\app\src\app_packages\win32com\__init__.py", line 8, in <module>
    import pythoncom
  File "G:\beeware-tutorial\bydkmipwd\build\bydkmipwd\windows\app\src\app_packages\pythoncom.py", line 2, in <module>
    import pywintypes
ModuleNotFoundError: No module named 'pywintypes'
Traceback (most recent call last):
  File "\app\bydkmipwd\__main__.py", line 1, in <module>
    from bydkmipwd.PwdUpdateToga import main
  File "\app\bydkmipwd\PwdUpdateToga.py", line 6, in <module>
    from bydkmipwd import TimeUtils, MacUtils, KeyUtils
  File "\app\bydkmipwd\MacUtils.py", line 2, in <module>
    import wmi
  File "\app_packages\wmi.py", line 105, in <module>
    from win32com.client import GetObject, Dispatch
  File "\app_packages\win32com\__init__.py", line 8, in <module>
    import pythoncom
  File "\app_packages\pythoncom.py", line 2, in <module>
    import pywintypes
**ModuleNotFoundError: No module named 'pywintypes'**

### Steps to reproduce

MacUtils.py line 2: import wmi
I have performed 'pip install pywin32' and 'pip install wmi' operations.
And I have referenced 'pywin32' and 'wmi' in the pyproject.toml file.

![image](https://github.com/beeware/beeware/assets/32702715/fe09e2fd-58e1-46a3-a57c-45f270b9c575)
![image](https://github.com/beeware/beeware/assets/32702715/f6f0098d-e6f9-434b-855a-50e577bf4ca9)

### Expected behavior

Now the 'run' operation reports the error mentioned above, but the 'dev' operation can start the project normally. I don't know where the problem lies, and I hope to get some clarification. Thank you.

### Environment

Date/Time:       2024-03-22 17:43:51 
Command line:    G:\beeware-tutorial\beeware-venv\Scripts\briefcase run

OS Release:      Windows 10
OS Version:      10.0.19045
Architecture:    AMD64
Platform:        Windows-10-10.0.19045-SP0

Python exe:      G:\beeware-tutorial\beeware-venv\Scripts\python.exe
Python version:  3.9.13 (tags/v3.9.13:6de2ca5, May 17 2022, 16:36:42) [MSC v.1929 64 bit (AMD64)]
Virtual env:     True
Conda env:       False

Briefcase:       0.3.15
Target platform: windows
Target format:   app

### Logs

```

```

**Comment by rmartin16** [comment]
When new requirements for an app are added in `pyproject.toml`, the build for the app must be updated. You can either run `briefcase update` or `briefcase run -r`.

**Reply by ycw01** [reply]
Thank you for your response, but I have tried both 'briefcase update' and 'briefcase run -r' commands, and neither resolved the issue. Additionally, I attempted to manually delete files under the 'build' directory, but still encountered the same error: ModuleNotFoundError: No module named 'pywintypes'.

**Reply by rmartin16** [reply]
Took another look; the `pywin32` package comes with a `pth` file and as such this issue is probably related to https://github.com/beeware/briefcase/issues/381.

**Comment by rmartin16** [comment]
~I was assuming `pywintypes` was a package on PyPI....but it appears it's simply a file named `pywintypes.py` in your app.~

~If you want to import another file/module in your project, you can use `import <app name>.pywintypes` or `from . import pywintypes` (where `<app name>` is replaced with the name of your app's package name).~

[edit] removed
